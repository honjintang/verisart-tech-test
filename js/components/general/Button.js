import React from 'react'

const Button = ({ text, onClickFn }) => (
    <button className="loginButton" onClick={onClickFn}>{text}</button>
)

export default Button