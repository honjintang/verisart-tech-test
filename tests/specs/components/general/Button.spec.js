import React from 'react'
import { shallow } from 'enzyme'
import Button from '~/components/general/Button'

const mockOnClick = jest.fn()

describe('components/display/Button', () => {
    it('matches snapshot', () => {
        const component = shallow(<Button onClickFn={mockOnClick} text='hello' />)
        expect(component).toMatchSnapshot()
    })
})