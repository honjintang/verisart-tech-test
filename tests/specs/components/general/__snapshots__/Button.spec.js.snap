// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/display/Button matches snapshot 1`] = `
<button
  className="loginButton"
  onClick={[MockFunction]}
>
  hello
</button>
`;
